.github/workflows/integration_test_run.yaml

```diff
@@ -204,6 +208,17 @@ jobs:
       - name: Pre-run script
         if: ${{ inputs.pre-run-script != '' }}
         run: bash ${{ inputs.pre-run-script }}
+      - uses: actions/cache@v3
+        id: cache-integration
+        with:
+          path: |
+            '**/*.charm'
+            .tox
+          key: integration-cache-${{ steps.get-date.outputs.date }}-${{ hashFiles('**/charmcraft.yaml', '**/metadata.yaml', '**/actions.yaml', '**/config.yaml', '**/src') }}
```

Comment:
Perhaps we should include the integration test dir too? It is my understanding that this charm is used to run those.